libs/game/console.ts

```diff
@@ -15,31 +15,40 @@ namespace game.consoleOverlay {
         consoleStrings = [];
     }
 
-    export function setVisible(value: boolean) {
+    export function setVisible(value: boolean, col?: number) {
         if (value != !!consoleStrings)
             consoleStrings = value ? [] : undefined;
+        if (col !== undefined)
+            consoleColor = col;
     }
 
     function listener(priority: ConsolePriority, text: string) {
         if (!consoleStrings)
             return;
-        else if (consoleStrings.length < consoleLines)
+
+        let newLineIndex = text.indexOf("\n");
```

Comment:
why are we not using "split"?

Reply:
Oops, must have just blanked for a bit, I did this right before leaving for the day. fixed